common/CommonAPI.h

```diff
@@ -734,23 +734,68 @@ class CommonAPI
 		uint32_t windowWidth = 800u;
 		uint32_t windowHeight = 600u;
 		uint32_t scImageCount = 3u;
-		
+
 		SFeatureRequest<nbl::video::IAPIConnection::E_FEATURE> requiredInstanceFeatures = {};
 		SFeatureRequest<nbl::video::IAPIConnection::E_FEATURE> optionalInstanceFeatures = {};
 		SFeatureRequest<nbl::video::ILogicalDevice::E_FEATURE> requiredDeviceFeatures = {};
 		SFeatureRequest<nbl::video::ILogicalDevice::E_FEATURE> optionalDeviceFeatures = {};
 		
-		nbl::asset::IImage::E_USAGE_FLAGS swapchainImageUsage = nbl::asset::IImage::E_USAGE_FLAGS::EUF_NONE;
-		nbl::core::vector<nbl::video::ISurface::SFormat> acceptableSurfaceFormats = {
-			nbl::video::ISurface::SFormat(nbl::asset::EF_R8G8B8A8_SRGB, nbl::asset::ECP_SRGB, nbl::asset::EOTF_sRGB),
-			nbl::video::ISurface::SFormat(nbl::asset::EF_B8G8R8A8_SRGB, nbl::asset::ECP_SRGB, nbl::asset::EOTF_sRGB)
+		nbl::asset::IImage::E_USAGE_FLAGS swapchainImageUsage = nbl::asset::IImage::E_USAGE_FLAGS::EUF_COLOR_ATTACHMENT_BIT;
+		nbl::core::vector<nbl::asset::E_FORMAT> acceptableSurfaceFormats = {
+			nbl::asset::EF_R8G8B8A8_SRGB,
+			nbl::asset::EF_R8G8B8A8_UNORM,
+			nbl::asset::EF_B8G8R8A8_SRGB,
+			nbl::asset::EF_B8G8R8A8_UNORM
 		};
+		nbl::core::vector<nbl::asset::E_COLOR_PRIMARIES> acceptableColorPrimaries = { nbl::asset::ECP_SRGB };
+		nbl::core::vector<nbl::asset::ELECTRO_OPTICAL_TRANSFER_FUNCTION> acceptableEotfs = { nbl::asset::EOTF_sRGB };
+		nbl::core::vector<nbl::video::ISurface::E_PRESENT_MODE> acceptablePresentModes = { nbl::video::ISurface::EPM_FIFO_RELAXED };
+		nbl::core::vector<nbl::video::ISurface::E_SURFACE_TRANSFORM_FLAGS> acceptableSurfaceTransforms = { 
+			nbl::video::ISurface::EST_IDENTITY_BIT,  
+			// (Flip vertical, used for OpenGL)
+			nbl::video::ISurface::EST_HORIZONTAL_MIRROR_ROTATE_180_BIT
+		};
+
 		nbl::asset::E_FORMAT depthFormat = nbl::asset::EF_UNKNOWN;
 
 		nbl::core::smart_refctd_ptr<nbl::ui::IWindow> window = nullptr;
 		nbl::core::smart_refctd_ptr<nbl::ui::IWindowManager> windowManager = nullptr;
 		nbl::core::smart_refctd_ptr<CommonAPIEventCallback> windowCb = nullptr;
 		nbl::core::smart_refctd_ptr<nbl::system::ILogger> logger = nullptr;
+
+		bool headlessCompute = false;
```

Comment:
no need for this bool, you should be able to deduce from requiring the swapchain & surface features